src/prepare.c

```diff
@@ -544,9 +544,7 @@ static inline void pthreads_prepare_functions(pthreads_object_t* thread) {
 			continue;
 		}
 
-		if (!(GC_FLAGS(name) & IS_STR_PERSISTENT)) {
```

Comment:
This is left overs from pthreads_global_string, which is no longer used.

You may remove that function from globals.* to complete the job there ;)